shoppingcart/workflow.go

```diff
@@ -0,0 +1,71 @@
+package shoppingcart
+
+import (
+	"fmt"
+	"go.temporal.io/sdk/workflow"
+)
+
+var (
+	UpdateName    = "shopping-cart"
+	TaskQueueName = "shopping-cart-tq"
+)
+
+type CartSignalPayload struct {
+	Action   string `json:"action"` // "add" or "remove"
+	ItemID   string `json:"item_id"`
+	Quantity int    `json:"quantity"`
+}
+
+type CartState struct {
+	Items map[string]int // itemID -> quantity
+}
+
+func CartWorkflow(ctx workflow.Context) error {
+	cart := CartState{make(map[string]int)}
+	logger := workflow.GetLogger(ctx)
+	var checkout bool
+
+	if err := workflow.SetUpdateHandlerWithOptions(ctx, UpdateName, func(ctx workflow.Context, actionType string, itemID string) (CartState, error) {
+		logger.Info("Received update,", actionType, itemID)
+		if actionType == "checkout" {
+			checkout = true
+		}
+		if itemID != "" {
```

Comment:
Does `itemID == ""` correspond to `checkout = true`? As in my other comment, I suggest making checkout a separate operation. (Use `switch` for `actionType`?)